capsules/src/debug_process_restart.rs

```diff
@@ -0,0 +1,59 @@
+//! Debug capsule to cause a button press to make all apps fault.
+//!
+//! This is useful for debugging that capsules and apps work when they are
+//! restarted by the kernel.
+//!
+//! Usage
+//! -----
+//!
+//! ```rust
+//! struct ProcessMgmtCap;
+//! unsafe impl capabilities::ProcessManagementCapability for ProcessMgmtCap {}
+//! let debug_process_restart = static_init!(
+//!     capsules::debug_process_restart::DebugProcessRestart<
+//!         'static,
+//!         sam4l::gpio::GPIOPin,
+//!         ProcessMgmtCap,
+//!     >,
+//!     capsules::debug_process_restart::DebugProcessRestart::new(
+//!         board_kernel,
+//!         &sam4l::gpio::PA[16],
+//!         ProcessMgmtCap
+//!     )
+//! );
+//! sam4l::gpio::PA[16].set_client(debug_process_restart);
+//! ```
+
+use kernel::capabilities::ProcessManagementCapability;
+use kernel::hil;
+use kernel::hil::gpio::{Client, InterruptMode};
+use kernel::Kernel;
+
+pub struct DebugProcessRestart<'a, G: hil::gpio::Pin + 'a, C: ProcessManagementCapability> {
+    kernel: &'static Kernel,
+    _pin: &'a G,
```

Comment:
Why store a reference that's never used? I think you should be able to remove this, which will let you remove the lifetime parameter for the struct and generally simplify this a bit.

Reply:
Updated.